ovc5/scripts/install_sd.sh

```diff
@@ -0,0 +1,273 @@
+  
+  # First, a FAT32 filesystem named boot in the first partition:
+  sudo mkfs.vfat -F 32 -n boot ${target_device}1
+  # Next, an ext4 filesystem named root in the second partition:
+  sudo mkfs.ext4 -L root ${target_device}2
+}
+
+umount_drive () {
+  # Unmount boot partition.
+  sudo umount $BOOT_DIR
+  sudo rmdir $BOOT_DIR
+
+  # Unmount root partition.
+  sudo umount $ROOT_DIR
+  sudo rmdir $ROOT_DIR
+
+  # Remove mount directory.
+  sudo rmdir $MOUNT_DIR
+}
+
+mount_drive () {
+  # Mount boot partition.
+  sudo mkdir -p $BOOT_DIR
+  sudo mount ${target_device}1 $BOOT_DIR
+
+  # Mount root partition.
+  sudo mkdir -p $ROOT_DIR
+  sudo mount ${target_device}2 $ROOT_DIR
+}
+
+copy_bin () {
+  sudo cp $BOOT_FILES_DIR/BOOT.BIN $BOOT_DIR/boot.bin
+  sudo cp $BOOT_FILES_DIR/image.ub $BOOT_DIR
+  sudo cp $BOOT_FILES_DIR/boot.scr $BOOT_DIR
+}
+
+install_debian () {
+  sudo apt install qemu-user-static debootstrap debian-archive-keyring schroot
+  sudo apt-key add /usr/share/keyrings/debian-archive-keyring.gpg
+  # Sets up debian buster arm port on the sd card
+  sudo qemu-debootstrap \
+    --arch=arm64 \
+    --keyring /usr/share/keyrings/debian-archive-keyring.gpg \
+    --variant=buildd buster \
+    $ROOT_DIR http://ftp.debian.org/debian
+}
+
+copy_petalinux_fs () {
+  # Get rootfs from petalinux
+  mkdir -p $TMP_MOUNT_DIR
+  gunzip -c $BOOT_FILES_DIR/rootfs.cpio.gz | sh -c 'cd /tmp/petalinux_mnt && cpio -i'
+  sudo cp $TMP_MOUNT_DIR/lib/modules $ROOT_DIR/lib -r
+  rm $TMP_MOUNT_DIR -rf
+}
+
+setup_userland () {
+  echo "[arm64-debian]
+description=Debian Buster (arm64)
+directory=$ROOT_DIR
+root-users=root
+users=root
+type=directory" | sudo tee /etc/schroot/chroot.d/arm64-debian
+
+  # Change the nssdatabases file to prevent copying of local userland.
+  for attribute in passwd shadow group services protocols networks hosts
+  do
+    sudo sed -e "/$attribute/ s/^#*/#/" -i /etc/schroot/default/nssdatabases
+  done
+
+interfaces_text="
+auto lo eth0
+allow-hotplug eth0
+iface lo inet loopback
+iface eth0 inet dhcp"
+
+  echo "
+sed -e \"s/$(hostname)/zynq/\" -i /etc/hosts
+echo zynq > /etc/hostname
+passwd
+$TEMP_PASSWORD
+$TEMP_PASSWORD
+apt update
+apt install -y vim locales openssh-server ifupdown net-tools iputils-ping avahi-autoipd avahi-daemon haveged i2c-tools rsyslog
+apt install -y git cmake libi2c-dev isc-dhcp-server
+grep -qxF 'ttyPS0' /etc/securetty || echo 'ttyPS0' >> /etc/securetty
+grep -qxF '$interfaces_text' /etc/network/interfaces || echo '$interfaces_text' >> /etc/network/interfaces
+egrep -v '^\s*#' /etc/ssh/sshd_config | grep -qxF 'PermitRootLogin yes' || echo 'PermitRootLogin yes' >> /etc/ssh/sshd_config
+egrep -v '^\s*#' /etc/ssh/sshd_config | grep -qxF 'PasswordAuthentication yes' || echo 'PasswordAuthentication yes' >> /etc/ssh/sshd_config
+" | sudo schroot -c arm64-debian -u root
+
+  echo "
+Execute and run through (likely using \"158. en_US.UTF-8 UTF-8\"):
+  dpkg-reconfigure locales
```

Comment:
Is this being done? I tried to install a package I needed and I got the following output during the process:

```
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LANG = "en_SG.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
```

Reply:
It opens a shell and tells you to do it manually. That's the only manual portion of the script. I didn't want to automate it in case someone wants to use a different language/locale.